integration/benchmarks/rax/run_benchmark.py

```
@@ -0,0 +1,740 @@
+        for t in threads:
+            t.join()
+        ingest_end = time.perf_counter()
+        ingest_duration = ingest_end - ingest_start
+        total_tokens = sum(len(title.split()) + len(body.split()) for _, title, _, body in docs)
+        token_rate = total_tokens / ingest_duration if ingest_duration > 0 else 0.0
+        
+        ingest_p50 = statistics.median(ingest_latencies) if ingest_latencies else 0.0
+        ingest_p95 = statistics.quantiles(ingest_latencies, n=20)[18] if len(ingest_latencies) >= 20 else ingest_p50
+        ingest_p99 = statistics.quantiles(ingest_latencies, n=100)[98] if len(ingest_latencies) >= 100 else ingest_p95
+
+        print(f"Ingestion completed in {ingest_duration:.2f}s | Throughput: {ingest_rate:,.1f} docs/s ({token_rate:,.1f} tokens/s) | Latency p50={ingest_p50:.2f}ms, p95={ingest_p95:.2f}ms, p99={ingest_p99:.2f}ms")
```

Ruff (0.16.1)

[error] Undefined name `ingest_rate`

(F821)